common/helpers/filterMatchScheduleEntry.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Filtering of schedule entries by various criteria.
+ * Only the matcher function for matching a single scheduleEntry with
+ * a set of filter criteria is implemented here, because this can be
+ * used more flexibly in the application.
+ */
+const filterMatchScheduleEntry = (scheduleEntry, filter) => {
+  if (!filter) return true
+  return (
+    // filter by period
+    (filter.period === null ||
+      filter.period === undefined ||
+      scheduleEntry.period()._meta.self === filter.period) &&
+    // filter by categories: OR filter
+    (filter.category === null ||
+      filter.category === undefined ||
+      filter.category.length === 0 ||
+      filter.category?.includes(
+        scheduleEntry.activity().category()._meta.self
+      )) &&
+    // filter by responsibles: AND filter
+    (filter.responsible === null ||
+      filter.responsible === undefined ||
+      filter.responsible.length === 0 ||
+      typeof filter.responsible?.every !== 'function' ||
+      filter.responsible?.every((responsible) => {
```

Comment:
```suggestion
      filter.responsible.every((responsible) => {
```

I would remove the `typeof` check. 
a) Function exists - then the check is not needed. 
b) Function does not exist - then the feature does not work.

As it is now, we can't determine if it would need a polyfill.

Reply:
Okay, fixed in [4014a33](https://github.com/ecamp/ecamp3/pull/7376/commits/4014a3356bdf9dbbfa2b640a2b3e64b8f19011cd). I had the function check there for the case that a non-array is passed in the responsible filter, which we currently don't support. This will throw an error without the function check. But it should surface quickly, should it ever happen.